File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    R = 0
    T = 0
    C = 0
    F = 0
    J = 0
    M = 0
    A = 0
    N = 0
    grade = {1:3,
            2:2,
            3:1,
            4:0,
            5:1,
            6:2,
            7:3}
    for i in range(len(survey)):
        if survey[i] == 'AN':
            if choices[i] <= 3:
                A += grade[choices[i]]
            elif choices[i] >=5:
                N += grade[choices[i]]
        elif survey[i] == 'NA':
            if choices[i] <= 3:
                N += grade[choices[i]]
            elif choices[i] >=5:
                A += grade[choices[i]]
        elif survey[i] == 'CF':
            if choices[i] <= 3:
                C += grade[choices[i]]
            elif choices[i] >=5:
                F += grade[choices[i]]
        elif survey[i] == 'FC':
            if choices[i] <= 3:
                F += grade[choices[i]]
            elif choices[i] >=5:
                C += grade[choices[i]]
        elif survey[i] == 'JM':
            if choices[i] <= 3:
                J += grade[choices[i]]
            elif choices[i] >=5:
                M += grade[choices[i]]
        elif survey[i] == 'MJ':
            if choices[i] <= 3:
                M += grade[choices[i]]
            elif choices[i] >=5:
                J += grade[choices[i]]
        elif survey[i] == 'RT':
            if choices[i] <= 3:
                R += grade[choices[i]]
            elif choices[i] >=5:
                T += grade[choices[i]]
        elif survey[i] == 'TR':
            if choices[i] <= 3:
                T += grade[choices[i]]
            elif choices[i] >=5:
                R += grade[choices[i]]
    if R >= T:
        answer += 'R'
    else:
        answer += 'T'
    if C >= F:
        answer += 'C'
    else:
        answer += 'F'
    if J >= M:
        answer += 'J'
    else:
        answer += 'M'
    if A >= N:
        answer += 'A'
    else:
        answer += 'N'
        
    return answer

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.HashMap;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        int[] score = new int[8];

        HashMap<Character,Integer> map = new HashMap<>();
        map.put('R',0);
        map.put('T',1);
        map.put('C',2);
        map.put('F',3);
        map.put('J',4);
        map.put('M',5);
        map.put('A',6);
        map.put('N',7);
        int[] character = new int[8];

        for(int i=0; i<survey.length; i++) {
            if(choices[i]==4) {
                continue;
            }
            if(choices[i] < 4) {
                score[map.get(survey[i].charAt(0))] += 4-choices[i];
            } else if (choices[i] > 4 ) {
                score[map.get(survey[i].charAt(1))] += choices[i]-4;
            }
        }
        if(score[0]==score[1]) answer+="R";
        else if(score[0]>score[1]) answer+="R";
        else answer+="T";

        if(score[2]==score[3]) answer+="C";
        else if(score[2]>score[3]) answer+="C";
        else answer+="F";

        if(score[4]==score[5]) answer+="J";
        else if(score[4]>score[5]) answer+="J";
        else answer+="M";

        if(score[6]==score[7]) answer+="A";
        else if(score[6]>score[7]) answer+="A";
        else answer+="N";

        return answer;
    }
}